src/middlewares/privilege.js

```diff
@@ -0,0 +1,28 @@
+const User = require("../models/user.model");
+
+const authorize = (privileges) => {
+  return async (req, res, next) => {
+    if (!req.session.currentUser) {
+      return res.redirect("/login");
+    }
+
+    const builtUser = new User(req.session.currentUser);
+    let userPrivileges = await builtUser.getPrivileges();
```

Comment:
You dont need to build a new user, you can get the db one
```suggestion
    const user =  User.getById(req.session.currentUser.uid);
    let userPrivileges = await user.getPrivileges();
```